test/integration/targets/nxos_pim_interface/tests/common/sanity.yaml

```diff
@@ -57,31 +57,96 @@
         - "result.changed == true"
 
   - name: Check idempotence
-    nxos_pim_interface: *config
+    nxos_pim_interface: *configjp
     register: result
 
   - assert: &false
       that:
         - "result.changed == false"
 
-  - name: configure gp policy and type
-    nxos_pim_interface: &configjp
+  - name: configure neighbor policy and rm
+    nxos_pim_interface: &confignpr
       interface: "{{ testint }}"
-      jp_policy_in: JPIN
-      jp_policy_out: JPOUT
-      jp_type_in: routemap
-      jp_type_out: routemap
+      neighbor_policy: NPR
+      neighbor_type: routemap
       provider: "{{ connection }}"
     register: result
 
   - assert: *true
 
   - name: Check idempotence
-    nxos_pim_interface: *configjp
+    nxos_pim_interface: *confignpr
+    register: result
+
+  - assert: *false
+
+  - pause:
+      seconds: 5
+
+  - name: configure neighbor policy and prefix
+    nxos_pim_interface: &confignpp
+      interface: "{{ testint }}"
+      neighbor_policy: NPPF
+      neighbor_type: prefix
+      provider: "{{ connection }}"
+    register: result
+
+  - assert: *true
+
+  - name: Check idempotence
+    nxos_pim_interface: *confignpp
     register: result
 
   - assert: *false
 
+  - name: configure hello_auth_key
+    nxos_pim_interface: &confighak1
+      interface: "{{ testint }}"
+      hello_auth_key: password1
+      provider: "{{ connection }}"
+    register: result
+
+  - assert: *true
+
+  - name: configure pim other params
+    nxos_pim_interface: &configo
+      interface: "{{ testint }}"
+      dr_prio: 10
+      hello_interval: 40
+      sparse: True
+      border: True
+      state: present
+      provider: "{{ connection }}"
+    register: result
+
+  - assert: *true
+
+  - name: Check idempotence
+    nxos_pim_interface: *configo
+    register: result
+
+  - assert: *false
+
+  - name: configure negative
+    nxos_pim_interface: &configno
+      interface: "{{ testint }}"
+      sparse: False
+      border: False
+      state: present
+      provider: "{{ connection }}"
+    register: result
+
+  - assert: *true
+
+  - name: Check idempotence
+    nxos_pim_interface: *configno
+    register: result
+
+  - assert: *false
+
+  - pause:
+      seconds: 5
```

Comment:
@saichint It would be great if you can add test for `state: absent` including idempotence.